src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs

```diff
@@ -73,10 +73,8 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             }
 
             (string exceptionType, string exceptionMessage, string exceptionDetails) = exception.GetExceptionDetails();
-            if (exception != null)
-            {
-                formattedMessage = Sanitizer.Sanitize(formattedMessage);
-            }
+
+            formattedMessage = Sanitizer.Sanitize(formattedMessage);
```

Comment:
I fear this is going to be a significant perf regression. Logging is expensive enough as is. But adding in scanning of every log for secrets, and then string modifications and allocations if there is? This will be a non-negligible perf impact.

Why are these secrets even being logged in the first place?

Reply:
HI @jviau 
According to the logging, the log was NOT sanitized at all. That means, the sanitizing was skipped. 
I don't have full context of this class; however, if you don't like this solution, how about when we don't have exception, we don't do any logging. (return  if exception == null) or any other good idea?

Reply:
Yes, this would be regressing work I did in https://github.com/Azure/azure-functions-host/commit/826c0b520c1e81d2f3106d08711c08f1bce4aa95 for throughput improvements that greatly increased our throughput by removing Sanitize from the hotpath. Sanitization should be done upstream as needed upstream by the component logging.

Reply:
I did the research of the logging pipeline. This pipeline is for Azure Monitor. 
So that customer logging should not be sanitized by us.
In this case, we should not sanitize in here. We are keeping on discussing on the internal e-mail thread. 
I'll abandoned this PR.